deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl

```diff
@@ -635,6 +635,29 @@ check_credentials(Username, Password, SslLoginName, PeerIp) ->
             {ok, {UserBin, PassBin}}
     end.
 
+%% Extract client_id from the certificate provided it was configured to do so and 
+%% it is possible to extract it else returns the client_id passed as parameter
+-spec extract_client_id_from_certificate(client_id(), rabbit_net:socket()) -> {ok, client_id()} | {error, reason_code()}.
+extract_client_id_from_certificate(Client0, Socket) ->
+    ClientIdPair = {Client0, case extract_ssl_cert_client_id_settings() of 
+        none -> Client0;
+        SslClientIdSettings -> 
+            case ssl_client_id(Socket, SslClientIdSettings) of 
+                none -> Client0;
+                V -> V 
+            end
+    end},
+    case ClientIdPair of 
+        {Client0, Client1} when Client0 == Client1 -> 
+            {ok, Client0};
+        {Client0, Client1} ->
+            ?LOG_ERROR(
+                "MQTT login failed: client_id in cert (~p) does not match client_id in protocol (~p)",
+                    [Client1, Client0]),
+            {error, ?RC_CLIENT_IDENTIFIER_NOT_VALID}
+    end.
```

Comment:
Simplify this function. Creating tuple `ClientIdPair` is unnecessary.

Reply:
indeed